include/tatami/isometric/arith_utils.hpp

```diff
@@ -45,6 +48,25 @@ void delayed_arith_run(Value_& val, Scalar_ scalar) {
         } else {
             val = scalar / val;
         }
+    } else if constexpr(op_ == DelayedArithOp::POWER) {
+        if constexpr(right_) {
+            val = std::pow(val, scalar);
+        } else {
+            val = std::pow(scalar, val);
+        }
+    } else if constexpr(op_ == DelayedArithOp::MODULO) {
+        if constexpr(right_) {
+            val = std::fmod(val, scalar);
+        } else {
+            val = std::fmod(scalar, val);
+        }
+    } else if constexpr(op_ == DelayedArithOp::INTEGER_DIVIDE) {
+        // x == (x %% y) + y * (x %/% y)
+        if constexpr(right_) {
+            val = (val - std::fmod(val, scalar)) / scalar;
```

Comment:
Couldn't we just do `std::trunc(val/scalar)`?

Reply:
`std::floor(x / y)` seems to do the trick

Reply:
Done